spec/features/plp_spec.rb

```diff
@@ -1,144 +1,104 @@
+      taxonomy_title = all('.taxonomy-root').map { |taxonomy| taxonomy.text.split.last }
+      title_is_present = taxonomy_title.any? { |word| breadcrumbs.include? word }
 
-        aggregate_failures do
-          expect(breadcrumbs).to include 'Home'
-          expect(breadcrumbs).to include 'Products'
-          expect(title_is_present).to eq true
-          expect(breadcrumbs).to include title
-        end
+      aggregate_failures do
+        expect(breadcrumbs).to include 'Home'
+        expect(breadcrumbs).to include 'Products'
+        expect(title_is_present).to eq true
+        expect(breadcrumbs).to include title
       end
     end
+  end
 
-    scenario 'can filter products by Categories' do
-      taxonomies = all('.list-group-item').collect(&:text)
-      taxonomies.each do |taxonomy|
-        click_link taxonomy
-        title = find('.taxon-title').text
-
-        sorted_products = all('span.info').collect(&:text)
-        sorted_products.each do |product|
-          visit router.admin_products_path
+  it_behaves_like 'Filtering by', 'Categories'
 
-          find('a', text: product).click
+  it_behaves_like 'Filtering by', 'Subcategories'
 
-          taxons = find('#s2id_product_taxon_ids')
-          within(taxons) { expect(page).to have_content taxonomy }
-        end
+  scenario 'Filtering by Price' do
+    admin_login
 
-        visit router.root_path
-      end
-    end
+    visit router.rails_category_path
 
-    scenario 'can filter products by Subcategories' do
-      visit router.plp_with_subcategories_path
+    products_prices = {}
 
-      subcategories = all('.breadcrumbs')
-      subcategories.each do |subcategory|
-        taxonomy = subcategory.text
-        subcategory.click
+    if page.has_css?('.page')
+      all('.page').each do
+        all('.panel-default').each do |item|
+          products_prices.update get_product_price(item)
 
-        sorted_products = all('span.info').collect(&:text)
-        sorted_products.each do |product|
-          visit router.admin_products_path
-          find('a', text: product).click
-          taxons = find('#s2id_product_taxon_ids')
-          within(taxons) { expect(page).to have_content taxonomy }
+          click_on 'Next' if page.has_css?('.next_page')
         end
-
-        visit router.plp_with_subcategories_path
+      end
+    else
+      all('.panel-default').each do |item|
+        products_prices.update get_product_price(item)
       end
     end
 
-    scenario 'can filter products by Price' do
-      visit router.rails_category_path
-
-      products_prices = { }
-
-      if page.has_css?('.page')
-        all('.page').each do
-          all('.panel-default').each do |item|
-            products_prices.update get_product_price(item)
-
-            click_link 'Next' if page.has_css?('.next_page')
-          end
-        end
+    prices = find('.navigation', text: 'Price Range').all('input').map(&:value)
```

Comment:
I've personally searched for input's label and it was `all('.nowrap').map(&:text)` but it's your call what to use